This patch includes Solaris portability fixes.

It fixes a Ghostscript issue with a page having dimensions of 1x1 pixel. The
upstream version generates the following error message and fails.

Error: /rangecheck in --getinterval--
Operand stack:
   ClippedLabel   ()   0   -1
Execution stack:
   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nost
ringval--   --nostringval--   --nostringval--   false   1   %stopped_push   1933   1   3   %oparray_pop   1932
   1   3   %oparray_pop   1917   1   3   %oparray_pop   1787   1   3   %oparray_pop   --nostringval--   %error
exec_pop   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval
--   --nostringval--   --nostringval--   --nostringval--   --nostringval--   --nostringval--   --nostringval--
   %loop_continue
Dictionary stack:
   --dict:745/1123(ro)(G)--   --dict:0/20(G)--   --dict:88/200(L)--   --dict:34/100(L)--   --dict:31/100(L)--
Current allocation mode is local
Current file position is 15757                                                                                GPL Ghostscript 10.05.1: Unrecoverable error, exit code 1
It also includes <signal.h> for the kill() and SIGTERM declarations used by
pdftoraster.
--- a/cupsfilters/ghostscript.c
+++ b/cupsfilters/ghostscript.c
@@ -1029,10 +1029,7 @@
       char output[31] = "";
       int pagecount;
       size_t bytes;
-      // Ghostscript runs too long on files converted from djvu files
-      // Using -dDEVICEWIDTHPOINTS -dDEVICEHEIGHTPOINTS params solves the
-      // problem
-      snprintf(gscommand, 65536, "%s -q -dNOPAUSE -dBATCH -sDEVICE=bbox -dDEVICEWIDTHPOINTS=1 -dDEVICEHEIGHTPOINTS=1 %s 2>&1 | grep -c HiResBoundingBox",
+      snprintf(gscommand, 65536, "%s -q -dNOPAUSE -dBATCH -sDEVICE=inkcov -sOutputFile=- %s | grep -c 'OK$'",
 	       CUPS_GHOSTSCRIPT, filename);
     
       FILE *pd = popen(gscommand, "r");
--- a/cupsfilters/pdftoraster.c
+++ b/cupsfilters/pdftoraster.c
@@ -25,6 +25,7 @@
 #include <math.h>
 #include <unistd.h>
 #include <sys/wait.h>
+#include <signal.h>
 #include <fcntl.h>
 #include <errno.h>
 
